### DISCUSSION TITLE
Feature Request - Modular Scrolling to improve preformance

### DISCUSSION THREAD

**Discussion by Slgoetz** [idiotWu/smooth-scrollbar#363] [Ideas]
I have been looking around but unable to find any thing built for Smooth-Scrollbar to create Modular scrolling so only the active section in view scroll.  This should help with performance of large pages.

**Comment by dolphin-wood** [comment]
Can you explain the _Modular Scrolling_ in more detail?

**Comment by artkrsk** [comment]
@idiotWu Applying translation to a long page container (with a lot of content) degrades the overall scrolling performance. Guys who develop this library https://locomotivemtl.github.io/locomotive-scroll/ have implemented "by section" translation so the only in-view section is being translated (you can see this feature in live on their demo page). That would be really awesome to have the similar feature in Smooth Scrollbar!

**Comment by dolphin-wood** [comment]
@artkrsk thanks for the clarification! However, I don't think it's easy to split a large scrollable area into several section: we need to measure all the elements inside and decide when to insert/remove a child element into/from the container, just like implementing a [virtual list](https://github.com/bvaughn/react-virtualized) but at a more general level. Instead of adding this feature into the library, I'd suggest you implement it yourselves in more specific scenarios.

**Comment by artkrsk** [comment]
@idiotWu smooth scrolling plugin doesn't have to do this split work. It's assumed that we already have our content prepared and sliced by `<section>` tags or in some other way. For example, in Locomotive Scroll they put `[data-scroll-section]` to define a section for the modular scrolling. So we just need to translate the section that is currently in-view. 

It's not about the specific scenarios, but about performance. At the moment Smooth Scrollbar works very good for the short pages. But when it comes to a long scrolling container, it's not that smooth anymore.
